src/main/java/com/jcabi/beanstalk/maven/plugin/AbstractBeanstalkMojo.java

```diff
@@ -259,6 +332,82 @@ protected boolean isGreen(final Environment env) {
     }
 
     /**
+     * Validate given text in YAML format.
+     *
+     * @param text YAML text
+     */
+    protected void validYaml(final String text) {
+        new Yaml().load(text);
+    }
+
+    /**
+     * Validate the war file.
+     *
+     * @param zip Zip file
+     * @throws org.apache.maven.plugin.MojoFailureException Thrown, if
+     *  validation fails.
+     */
+    protected void validate(final ZipFile zip) throws MojoFailureException {
+        if (zip.getEntry(".ebextension") == null) {
+            throw new MojoFailureException(
+                ".ebextensions directory does not exist in the WAR file"
+            );
+        }
+        final Enumeration<? extends ZipEntry> entries = zip.entries();
+        while (entries.hasMoreElements()) {
+            final ZipEntry entry = entries.nextElement();
+            if (entry.getName().startsWith(".ebextensions/")
+                && !entry.isDirectory()) {
+                final String text = this.readFile(zip, entry);
```

Comment:
@tmjee this is also a redundant variable: http://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html